include/dali/core/tensor_shape.h

```diff
@@ -637,6 +638,32 @@ struct TensorListShapeBase {
     shapes.resize(num_samples * sample_dim);
   }
 
+
+  /**
+   * Append this TensorListShape with shapes from other TensorListShapes.
+   */
+  void concatenate(const std::vector<TensorListShape<sample_ndim>> &tlss) {
+    int new_samples = std::accumulate(tlss.begin(), tlss.end(), 0,
+                                      [](int acc, const auto &tls) -> int {
+                                          return acc + tls.num_samples();
+                                      });
+    if (new_samples == 0) return;
+    for (const auto &tls : tlss) {
+      DALI_ENFORCE(tls.sample_dim() == tlss[0].sample_dim(),
+                   "All samples in TensorListShape must have the same number of dimensions");
```

Comment:
Use std::logic_error instead - DALI_ENFORCE is not used in this file and that's on purpose.

Reply:
Done